File: internal/api/handlers/chatbot.go
package handlers

import (
	"context"
	"database/sql"

	"github.com/onurceri/botla-co/internal/models"
	"github.com/onurceri/botla-co/internal/services"
	"github.com/onurceri/botla-co/pkg/config"
)

// VectorStore interface for vector operations
type VectorStore interface {
	DeleteBySourceID(ctx context.Context, sourceID string) error
}

// ChatbotHandlers handles chatbot-related HTTP endpoints
type ChatbotHandlers struct {
	DB               *sql.DB
	Cfg              *config.Config
	VectorStore      VectorStore
	ChatbotService   *services.ChatbotService
	OrgService       *services.OrganizationService
	WorkspaceService *services.WorkspaceService
}

type createChatbotRequest struct {
	Name                 string                   `json:"name"`
	Description          *string                  `json:"description"`
	CustomInstruction    *string                  `json:"custom_instruction"`
	Language             *string                  `json:"language"`
	Model                *string                  `json:"model"`
	Temperature          *float32                 `json:"temperature"`
	MaxTokens            *int                     `json:"max_tokens"`
	ThemeColor           *string                  `json:"theme_color"`
	WelcomeMessage       *string                  `json:"welcome_message"`
	Position             *string                  `json:"position"`
	BotMessageColor      *string                  `json:"bot_message_color"`
	UserMessageColor     *string                  `json:"user_message_color"`
	BotMessageTextColor  *string                  `json:"bot_message_text_color"`
	UserMessageTextColor *string                  `json:"user_message_text_color"`
	ChatFontFamily       *string                  `json:"chat_font_family"`
	ChatHeaderColor      *string                  `json:"chat_header_color"`
	ChatHeaderTextColor  *string                  `json:"chat_header_text_color"`
	ChatBackgroundColor  *string                  `json:"chat_background_color"`
	BubbleRadius         *string                  `json:"bubble_radius"`
	InputBackgroundColor *string                  `json:"input_background_color"`
	InputTextColor       *string                  `json:"input_text_color"`
	SendButtonColor      *string                  `json:"send_button_color"`
	BotIcon              *string                  `json:"bot_icon"`
	BotDisplayName       *string                  `json:"bot_display_name"`
	SecureEmbedEnabled   *bool                    `json:"secure_embed_enabled"`
	AllowedDomains       *string                  `json:"allowed_domains"`
	EmbedSecret          *string                  `json:"embed_secret"`
	SuggestedQuestions   *[]string                `json:"suggested_questions"`
	SuggestionsEnabled   *bool                    `json:"suggestions_enabled"`
	IncludePaths         *[]string                `json:"include_paths"`
	ExcludePaths         *[]string                `json:"exclude_paths"`
	SelectorWhitelist    *[]string                `json:"selector_whitelist"`
	DiscoveryMode        *string                  `json:"discovery_mode"`
	RefreshPolicy        *string                  `json:"refresh_policy"`
	RefreshFrequency     *string                  `json:"refresh_frequency"`
	HideBranding         *bool                    `json:"hide_branding"`
	CustomBranding       *models.CustomBranding   `json:"custom_branding"`
	ConfidenceThreshold  *float64                 `json:"confidence_threshold"`
	FallbackMessages     *models.FallbackMessages `json:"fallback_messages"`
	TopicRestrictions    *models.TopicConfig      `json:"topic_restrictions"`
	HandoffEnabled       *bool                    `json:"handoff_enabled"`
	HandoffType          *string                  `json:"handoff_type"`
	HandoffConfig        *models.HandoffConfig    `json:"handoff_config"`
	ThresholdConfig      *models.ThresholdConfig  `json:"threshold_config"`
}

File: frontend/src/types/chatbot.ts
/**
 * Chatbot request types for mutations.
 * These types mirror the backend Go structs from internal/services/chatbot_service.go
 */

// Re-export shared types from useChatbotForm to avoid duplication
export type {
  CustomBranding,
  FallbackMessages,
  TopicConfig,
  HandoffConfig,
  ThresholdConfig,
} from '@/features/chatbot/hooks/useChatbotForm'

// Request types for update mutations
export interface ChatbotUpdateRequest {
  name?: string;
  description?: string | null;
  custom_instruction?: string | null;
  language?: string | null;
  model?: string | null;
  temperature?: number | null;
  max_tokens?: number | null;
  theme_color?: string | null;
  welcome_message?: string | null;
  position?: string | null;
  bot_message_color?: string | null;
  user_message_color?: string | null;
  bot_message_text_color?: string | null;
  user_message_text_color?: string | null;
  chat_font_family?: string | null;
  chat_header_color?: string | null;
  chat_header_text_color?: string | null;
  chat_background_color?: string | null;
  bubble_radius?: string | null;
  input_background_color?: string | null;
  input_text_color?: string | null;
  send_button_color?: string | null;
  bot_icon?: string | null;
  bot_display_name?: string | null;
  secure_embed_enabled?: boolean | null;
  allowed_domains?: string[];
  embed_secret?: string | null;
  suggested_questions?: string[] | null;
  manual_questions?: string[] | null;
  suggestions_enabled?: boolean | null;
  include_paths?: string[] | null;
  exclude_paths?: string[] | null;
  selector_whitelist?: string[] | null;
  discovery_mode?: string | null;
  refresh_policy?: string | null;
  refresh_frequency?: string | null;
  hide_branding?: boolean | null;
  custom_branding?: import('@/features/chatbot/hooks/useChatbotForm').CustomBranding | null;
  confidence_threshold?: number | null;
  fallback_messages?: import('@/features/chatbot/hooks/useChatbotForm').FallbackMessages | null;
  topic_restrictions?: import('@/features/chatbot/hooks/useChatbotForm').TopicConfig | null;
  threshold_config?: import('@/features/chatbot/hooks/useChatbotForm').ThresholdConfig | null;
  handoff_enabled?: boolean | null;
  handoff_type?: string | null;
  handoff_config?: import('@/features/chatbot/hooks/useChatbotForm').HandoffConfig | null;
}

export interface BasicInfoRequest {
  name: string;
  description?: string | null;
  language?: string | null;
  custom_instruction?: string | null;
}

export interface AppearanceRequest {
  theme_color?: string | null;
  welcome_message?: string | null;
  position?: string | null;
  bot_message_color?: string | null;
  user_message_color?: string | null;
  bot_message_text_color?: string | null;
  user_message_text_color?: string | null;
  chat_font_family?: string | null;
  chat_header_color?: string | null;
  chat_header_text_color?: string | null;
  chat_background_color?: string | null;
  bubble_radius?: string | null;
  input_background_color?: string | null;
  input_text_color?: string | null;
  send_button_color?: string | null;
  bot_icon?: string | null;
  bot_display_name?: string | null;
  hide_branding?: boolean | null;
  custom_branding?: import('@/features/chatbot/hooks/useChatbotForm').CustomBranding | null;
  suggested_questions?: string[] | null;
  manual_questions?: string[] | null;
  suggestions_enabled?: boolean | null;
}

export interface ModelSettingsRequest {
  model?: string | null;
  temperature?: number | null;
  max_tokens?: number | null;
  custom_instruction?: string | null;
}

export interface SecuritySettingsRequest {
  secure_embed_enabled?: boolean | null;
  allowed_domains?: string[];
  embed_secret?: string | null;
}

export interface GuardrailsRequest {
  confidence_threshold?: number | null;
  fallback_messages?: import('@/features/chatbot/hooks/useChatbotForm').FallbackMessages | null;
  topic_restrictions?: import('@/features/chatbot/hooks/useChatbotForm').TopicConfig | null;
  threshold_config?: import('@/features/chatbot/hooks/useChatbotForm').ThresholdConfig | null;
}

export interface HandoffRequest {
  handoff_enabled?: boolean | null;
  handoff_type?: string | null;
  handoff_config?: import('@/features/chatbot/hooks/useChatbotForm').HandoffConfig | null;
}

export interface RefreshRequest {
  refresh_policy?: string | null;
  refresh_frequency?: string | null;
}

export interface ScrapingConfigRequest {
  include_paths?: string[] | null;
  exclude_paths?: string[] | null;
  selector_whitelist?: string[] | null;
  discovery_mode?: string | null;
}

export interface CreateChatbotRequest {
  name: string;
  description?: string | null;
  custom_instruction?: string | null;
  language?: string | null;
  model?: string | null;
  temperature?: number | null;
  max_tokens?: number | null;
  theme_color?: string | null;
  welcome_message?: string | null;
  position?: string | null;
  bot_message_color?: string | null;
  user_message_color?: string | null;
  bot_message_text_color?: string | null;
  user_message_text_color?: string | null;
  chat_font_family?: string | null;
  chat_header_color?: string | null;
  chat_header_text_color?: string | null;
  chat_background_color?: string | null;
  bubble_radius?: string | null;
  input_background_color?: string | null;
  input_text_color?: string | null;
  send_button_color?: string | null;
  bot_icon?: string | null;
  bot_display_name?: string | null;
  secure_embed_enabled?: boolean | null;
  allowed_domains?: string;
  embed_secret?: string | null;
  suggested_questions?: string[] | null;
  suggestions_enabled?: boolean | null;
  include_paths?: string[] | null;
  exclude_paths?: string[] | null;
  selector_whitelist?: string[] | null;
  discovery_mode?: string | null;
  refresh_policy?: string | null;
  refresh_frequency?: string | null;
  hide_branding?: boolean | null;
  custom_branding?: import('@/features/chatbot/hooks/useChatbotForm').CustomBranding | null;
  confidence_threshold?: number | null;
  fallback_messages?: import('@/features/chatbot/hooks/useChatbotForm').FallbackMessages | null;
  topic_restrictions?: import('@/features/chatbot/hooks/useChatbotForm').TopicConfig | null;
  handoff_enabled?: boolean | null;
  handoff_type?: string | null;
  handoff_config?: import('@/features/chatbot/hooks/useChatbotForm').HandoffConfig | null;
  threshold_config?: import('@/features/chatbot/hooks/useChatbotForm').ThresholdConfig | null;
}
